### PR TITLE
chore: add new fields to notification class (cost_in_pounds, is_data_ready and cost_details)

### PR DESCRIPTION
**Pull request by joybytes**
## What problem does the pull request solve?
New fields have been added to the response of `get_notification(notificationId)` endpoint
- cost_in_pounds
- is_cost_data_ready
- cost_details:
-- **Nested fields for letters**:
--- billable_sheets_of_paper
--- postage
-- **Nested fields for sms**:
--- billable_sms_fragments
--- international_rate_multiplier
--- sms_rate
 
Based on the addition this PR updates our client documentation to showcase an aligned response for the endpoint

### Related task : 
[3. Add GET notification cost data to API clients and API client docs](https://trello.com/c/5VZrBNpU/845-3-add-get-notification-cost-data-to-api-clients-and-api-client-docs)

## Checklist

- [x] I’ve used the pull request template
- [x] I’ve written unit tests for these changes
- [x] I’ve updated the documentation in
  - [x] [notifications-tech-docs repository](https://github.com/alphagov/notifications-tech-docs/blob/main/source/documentation/client_docs/_java.md)
  - [x] `CHANGELOG.md`
- [x] I’ve bumped the version number
    - [x] in `src/main/resources/application.properties`
    - [x] in `pom.xml`

### CODE REVIEW

File: src/test/resources/v2_notifications_response.json [hunk 1]
```diff
@@ -11,7 +11,6 @@
       "line_4": "a line 4",
       "line_5": "a line 5",
       "line_6": "a line 6",
-      "line_7": "a line 7",
```
Comment (whpearson) [inline src/test/resources/v2_notifications_response.json:14]: Why was this removed?

File: src/main/java/uk/gov/service/notify/Notification.java [hunk 2]
```diff
@@ -21,16 +21,29 @@ public class Notification {
...
+    private UUID templateId;
+    private int templateVersion;
+    private String templateUri;
+
+    // CostDetails fields
+    private Integer billableSmsFragments;
+    private Double internationalRateMultiplier;
+    private Double smsRate;
+    private Integer billableSheetsOfPaper;
+    private String costPostage;
```
Comment (whpearson) [inline src/main/java/uk/gov/service/notify/Notification.java:46]: Is there a reason this is costPostage? I think postageType or postageRate might be better if it is "first class" but if it is for consistency, then that is okay.

Comment (joybytes) [reply src/main/java/uk/gov/service/notify/Notification.java:46]: I didn't know what to call it really.
The field name is `postage` but the Notification class already has a postage field too, and since we use a flat structure, the names would clash. I needed to make a distinction between the postage from Notification and nested postage from CostDetails.

I like postageType so I'll change to it

File: src/test/resources/v2_notifications_response.json (hunk 1 above)
Comment (joybytes) [reply src/test/resources/v2_notifications_response.json:14]: I removed it because we don't actually return that line as far as I am aware.
While testing different client-libraries, I kept seeing line_7 but I did not see the mention on the response and I did some digging and it seems that we do not return line_7 [here](https://github.com/alphagov/notifications-api/blob/7730ed50a32f43f8d119009c23b7f0d8eb6c3360/app/models.py#L1612)

File: src/main/java/uk/gov/service/notify/Notification.java (hunk 2 above)
Comment (joybytes) [reply src/main/java/uk/gov/service/notify/Notification.java]: https://github.com/alphagov/notifications-java-client/pull/244/commits/975e56c85caac90d464d2e1a4e136bfce38fbb3c

Comment (joybytes) [reply src/main/java/uk/gov/service/notify/Notification.java]: I also updated the documentation: https://github.com/alphagov/notifications-tech-docs/pull/243/commits/578dc7fc940f6713566b1a446b0a610c0600c55c

**Review by whpearson — APPROVED** [review]
Looks good
